nntrainer/src/tensor.cpp

```diff
@@ -224,24 +224,59 @@ Tensor Tensor::add(float const &value) { CLONE_OP_I(add_i, value); }
  * TODO: add axis rather doing add over the last two dimensions always
  */
 int Tensor::add_i(Tensor const &m, float const alpha) {
-  if ((dim.height() != m.dim.height()) || (dim.width() != m.dim.width())) {
+
+  ExternalLoopInfo e;
+  try {
+    e = this->computeExternalLoop(m);
+  } catch (std::exception &e) {
+    ml_loge("%s %s", typeid(e).name(), e.what());
     return ML_ERROR_INVALID_PARAMETER;
   }
 
-  float *data = getData();
-  const float *mdata = m.getData();
-  unsigned int len = length();
+  float *buf = this->getData();
+  const float *mbuf = m.getData();
 
-  unsigned int size = dim.getFeatureLen();
-  if (m.dim.batch() == 1) {
-    for (unsigned int k = 0; k < dim.batch(); ++k) {
-      saxpy(size, alpha, mdata, 1, &(data[k * size]), 1);
-    }
+  unsigned int offset = 0;
+  unsigned int m_offset = 0;
+
+  if (e.buffer_axis == -1) {
+    saxpy(e.buffer_size, alpha, mbuf, e.strides[3], buf, strides[3]);
   } else {
-    if (dim.batch() != m.dim.batch()) {
-      return ML_ERROR_INVALID_PARAMETER;
+    for (unsigned int b = 0; b < batch(); ++b) {
```

Comment:
find the first n axis till which stride matches,
then loop over n axis in 1-D fashion
and then call add_i over the new data itself.
The current method is not scalable to multiple operations IMO.

Reply:
I agree that the implementaion could be improved. Could you explain a bit more in person?

Reply:
As discussed offline, lets update to a more manageable implementation.